homeassistant/components/volvooncall/device_tracker.py

```diff
@@ -1,65 +1,68 @@
 """Support for tracking a Volvo."""
 from __future__ import annotations
 
-from homeassistant import config_entries
+from volvooncall.dashboard import Instrument
+
 from homeassistant.components.device_tracker import SourceType
 from homeassistant.components.device_tracker.config_entry import TrackerEntity
-from homeassistant.core import HomeAssistant
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from . import DOMAIN, VolvoEntity, VolvoUpdateCoordinator
+from . import VolvoData, VolvoEntity, VolvoUpdateCoordinator
+from .const import DOMAIN, VOLVO_DISCOVERY_NEW
 
 
 async def async_setup_entry(
     hass: HomeAssistant,
-    config_entry: config_entries.ConfigEntry,
+    config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Configure device_trackers from a config entry created in the integrations UI."""
-    volvo_data = hass.data[DOMAIN][config_entry.entry_id].volvo_data
-    for instrument in volvo_data.instruments:
-        if instrument.component == "device_tracker":
-            discovery_info = (
-                instrument.vehicle.vin,
-                instrument.component,
-                instrument.attr,
-                instrument.slug_attr,
-            )
-
-            async_add_entities(
-                [
+    coordinator: VolvoUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    volvo_data: VolvoData = coordinator.volvo_data
```

Comment:
We don't need this type as the coordinator is typed already including its attributes.
```suggestion
    volvo_data = coordinator.volvo_data
```